CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/utility/SaveResultsFileWriterNGTest.java

```diff
@@ -53,48 +82,152 @@ public SaveResultsFileWriterNGTest() {
+        try {
+            SaveResultsFileWriter.writeRecordStore(plugin, tabularRecordStore);
+
+            // Execution should not make it this far, an invalid file should result in an exception.
+            Assert.assertTrue(false, "Exception is expected.");
 
+        } catch (Exception ex) {
+            mockedRecordStoreUtilities.verify(times(0), () -> RecordStoreUtilities.toCsv(Mockito.any(), Mockito.any()));
+            Assert.assertFalse(fileCreated, "Record store file was not created.");
+            Assert.assertEquals(constellationLoggerHelperStatus, ConstellationLoggerHelper.FAILURE,
+                    "ConstellationLoggerHelper passed status = FAILURE.");
+            Assert.assertTrue(true, "Exception is expected.");
+        }
+    }
+   
     /**
-     * Test of generateFilename method, of class SaveResultsFileWriter.
+     * Test of writeRecordStore method for case when DataAccessPreferenceKeys.getDataAccessResultsDir() returns a
+     * valid directory. Confirm file is created with expected contents and that no exception is thrown. confirm
+     * ConstellationLoggerHelper.exportPropertyBuilder called with appropriate SUCCESS status.
      */
     @Test
-    public void testGenerateFilenameSuffix2() {
+    public void testWriteRecordStore() {
         Plugin plugin = new ExampleClass();
-        String filename = SaveResultsFileWriter.generateFilename(plugin, "xml");
-        Assert.assertFalse(filename.endsWith("..xml"));
+        TabularRecordStore tabularRecordStore = new TabularRecordStore();
+        String key = "TEST1KEY";
+        String value = "TEST1VALUE";
+        tabularRecordStore.add();
+        tabularRecordStore.set(key, value);
+
+        // This test actually creates the output file in user.home directory, checksa its contents, and then removes it.
+        mockedDataAccessPreferenceKeys.when(DataAccessPreferenceKeys::getDataAccessResultsDir).thenReturn(new File(System.getProperty("user.home")));
+        mockedRecordStoreUtilities.when(() -> RecordStoreUtilities.toCsv(Mockito.any(), Mockito.any())).thenCallRealMethod();
+        mockedConstellationLoggerHelper.when(() -> ConstellationLoggerHelper.exportPropertyBuilder(any(), any(), any(), any())).thenAnswer((var invocation) -> { 
+            Object[] args = invocation.getArguments();
+            File passedFile = (File)args[2];
+            constellationLoggerHelperStatus = (String)args[3];
+
+            // Ensure the file was created
+            fileCreated = passedFile.exists();
+            if (fileCreated) {
+                // Read file and confirm it contains 3 rows
+                int rows = 0;
+                Scanner reader = new Scanner(passedFile);
+                while (reader.hasNextLine()) {
+                    rows++;
+                    String data = reader.nextLine();
+                    if (rows == 1) dataValid = dataValid & (data.equals(key));
+                    if (rows == 2) dataValid = dataValid & (data.equals(value));
```

Comment:
I think `&&` is generally better to use than `&` for booleans. Also unsure how I feel about the inline if code block

Reply:
I've flicked the &/&& operator, I think & should work, but && probably more targetted
Ive added {} around blocks, I liked it inline as all the conditions 'lined up' vertically, but i'm happy to go with reccomendation.